src/OSPSuite.Infrastructure.Import/Services/Importer.cs

```diff
@@ -113,10 +119,6 @@ public void CalculateFormat(IDataSourceFile dataSource, IReadOnlyList<ColumnInfo
             throw new UnsupportedFormatException(dataSource.Path);
 
          dataSource.AvailableFormats = AvailableFormats(dataSource.DataSheets[sheetName].RawData, columnInfos, metaDataCategories).ToList();
```

Comment:
Looking at this code, we are modifying the dataSource given as parameter. Can I suggest to change this methods so that it does not have side effect and returns the availableFormats instead? Line 108 for instance . It is not clear at all that the format will be set in this method. Thougts?

Reply:
@abdelr not sure if you saw this

Reply:
> @abdelr not sure if you saw this

Sure, but this is still a work in progress since the last discussion on markers for the sheets